bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ExperienceFragmentImpl.java

```diff
@@ -0,0 +1,221 @@
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.*;
+import org.jetbrains.annotations.NotNull;
+
+import com.adobe.cq.export.json.ComponentExporter;
+import com.adobe.cq.export.json.ExporterConstants;
+import com.adobe.cq.wcm.core.components.models.ExperienceFragment;
+import com.day.cq.wcm.api.LanguageManager;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.Template;
+import com.day.cq.wcm.api.WCMException;
+import com.day.cq.wcm.msm.api.LiveCopy;
+import com.day.cq.wcm.msm.api.LiveRelationship;
+import com.day.cq.wcm.msm.api.LiveRelationshipManager;
+import com.day.text.Text;
+
+@Model(adaptables = SlingHttpServletRequest.class,
+    adapters = {ExperienceFragment.class, ComponentExporter.class },
+    resourceType = {ExperienceFragmentImpl.RESOURCE_TYPE_V1 })
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME,
+    extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class ExperienceFragmentImpl implements ExperienceFragment {
+
+    public static final String RESOURCE_TYPE_V1 = "core/wcm/components/experiencefragment/v1/experiencefragment";
+
+    private static final String PATH_DELIMITER = "/";
+    private static final char PATH_DELIMITER_CHAR = '/';
+    private static final String CONTENT_ROOT = "/content";
+    private static final String EXPERIENCE_FRAGMENTS_ROOT = "/content/experience-fragments";
+    private static final String JCR_CONTENT_ROOT = "/jcr:content";
+
+    @Self
+    private SlingHttpServletRequest request;
+
+    @Inject
+    protected Resource resource;
+
+    @SlingObject
+    private ResourceResolver resolver;
+
+    @ScriptVariable
+    private Page currentPage;
+
+    @ValueMapValue(name = ExperienceFragment.PN_FRAGMENT_PATH, injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String fragmentPath;
+
+    @OSGiService
+    private LanguageManager languageManager;
+
+    @OSGiService
+    private LiveRelationshipManager relationshipManager;
+
+    private String localizedFragmentPath;
+
+    @PostConstruct
+    protected void initModel() {
+        if (inTemplate()) {
+            String currentPageRootPath = getRoot(currentPage.getPath());
+            // we should use getRoot instead of getXfRoot once the XF UI supports creating Live and Language Copies
+            String xfRootPath = getXfRoot(fragmentPath, currentPageRootPath);
+            if (!StringUtils.isEmpty(currentPageRootPath) && !StringUtils.isEmpty(xfRootPath)) {
+                String xfRelativePath = StringUtils.substring(fragmentPath, xfRootPath.length());
+                String localizedXfRootPath = StringUtils.replace(currentPageRootPath, CONTENT_ROOT, EXPERIENCE_FRAGMENTS_ROOT, 1);
+                localizedFragmentPath = StringUtils.join(localizedXfRootPath, xfRelativePath, JCR_CONTENT_ROOT);
+            }
+        }
+        String xfContentPath = StringUtils.join(fragmentPath, JCR_CONTENT_ROOT);
```

Comment:
should we consider a case when an XF is placed on a page level? if not the whole logic of that method could be placed inside `isTemplate` condition